src/plugins/workspace/server/utils.ts

```diff
@@ -89,3 +90,26 @@ export const getOSDAdminConfigFromYMLConfig = async (
 
   return [groupsResult, usersResult];
 };
+
+export const getDataSourcesList = (client: SavedObjectsClientContract, workspaces: string[]) => {
+  return client
+    .find({
+      type: 'data-source',
+      fields: ['id', 'title'],
+      perPage: 10000,
+      workspaces,
+    })
+    .then((response) => {
```

Comment:
shall we handle exception here? if user don't have permission to list all data sources

Reply:
In new design only OSD admin will call this function and there will be a new PR to return empty array if user doesn't have permission. So i think we don't need to handle exception here.